waspc/data/Generator/templates/server/src/jobs/core/pgBoss/pgBossJob.js

```diff
@@ -68,17 +69,27 @@ class PgBossJob extends Job {
  * functions, we will override the previous calls.
  * @param {string} jobName - The user-defined job name in their .wasp file.
  * @param {fn} jobFn - The user-defined async job callback function.
- * @param {object} defaultJobOptions - PgBoss specific options for boss.send() applied to every submit() invocation,
+ * @param {object} defaultJobOptions - pg-boss specific options for boss.send() applied to every submit() invocation,
  *                                     which can overriden in that call.
+ * @param {object} jobSchedule [Optional] - The cron string and arguments/options when invoking the job.
  */
-export async function createJob({ jobName, jobFn, defaultJobOptions } = {}) {
-  // As a safety precaution against undefined behavior of registering different
-  // functions for the same job name, remove all registered functions first.
-  await boss.offWork(jobName)
+export async function createJob({ jobName, jobFn, defaultJobOptions, jobSchedule } = {}) {
+  pgBossStarted.then(async (boss) => {
+    // As a safety precaution against undefined behavior of registering different
+    // functions for the same job name, remove all registered functions first.
+    await boss.offWork(jobName)
 
-  // This tells pgBoss to run given worker function when job/payload with given job name is submitted.
-  // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#work
-  await boss.work(jobName, jobFn)
+    // This tells pgBoss to run given worker function when job/payload with given job name is submitted.
+    // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#work
+    await boss.work(jobName, jobFn)
+
+    // If a job schedule is provided, we should schedule the recurring job.
+    // If the schedule name already exists, it's updated to the provided cron expression, arguments, and options.
+    // Ref: https://github.com/timgit/pg-boss/blob/master/docs/readme.md#scheduling
+    if (jobSchedule && jobSchedule.cron) {
+      await boss.schedule(jobName, jobSchedule.cron, jobSchedule.performFnArg || null, jobSchedule.options || {})
+    }
+  })
```

Comment:
Why is this function `async`? Do we want to await until `pgBossStarted` before returning a `PgBossJob` instance?
1. If yes, it seems like we're missing an `await` statement before `pgBossStarted.then(...)`
2. If not, the function doesn't need to be `async` nor does it have to return a `Promise`

Minimal examples to run in the console:
```javascript
const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))

// Option 1
function createJob1() {
    sleep(1000).then(async () => { 
        console.log('1');
        await sleep(1000);
        console.log('2');
    })
    return "job";
}

const job = createJob1();

// Option 2
async function createJob2() {
    await sleep(1000).then(async () => { 
        console.log('1');
        await sleep(1000);
        console.log('2');
    })
    return "job";
}

const job = await createJob2();
```
Perhaps we don't require `createJob` to be `async` for _this particular executor_, but you decided to make it `async` anyway for the purposes of future executors (which might have to do something asynchronous before returning a job instance). If that's the case, we still don't need to return a Promise from `createJob`. This is because it's perfectly valid to [await a non-Promise value](https://stackoverflow.com/questions/55262996/does-awaiting-a-non-promise-have-any-detectable-effect):
```javascript
// This code works
function createJob1() {
    sleep(1000).then(async () => { 
        console.log('1');
        await sleep(1000);
        console.log('2');
    })
    return "job";
}

const job = await createJob1();
```

Reply:
Good observation (and examples!). It is case (2), and a minor oversight whereby before I wrapped that code into a then, I was awaiting from within the function itself for something. We can nix the `async` on the function itself, thanks! 👍🏻